mlir_graphblas/src/lib/GraphBLAS/GraphBLASPasses.cpp

```diff
@@ -474,7 +477,8 @@ class LowerMatrixApplyRewrite : public OpRewritePattern<graphblas::MatrixApplyOp
     Value c1 = rewriter.create<ConstantIndexOp>(loc, 1);
 
     // Get sparse tensor info
-    Value output = callDupTensor(rewriter, module, loc, inputTensor);
+    Value duplicate = callDupTensor(rewriter, module, loc, inputTensor);
+    Value output = convertToExternalCSR(rewriter, module, loc, duplicate);
```

Comment:
Should `dup` return a CSX? I understand that behavior for `empty_like` to return CSX, but I feel like `dup` should keep the same type as the input. I think MatrixSelect is the only op which calls `dup` and then resizes pointers.

Reply:
I was concerned that it would be unnecessarily restrictive and wouldn't let us call dup on a CSR and CSC matrix in the same module. If that's not a sufficiently motivating concern, I can make it return something of whatever type the input is.

Reply:
Let's leave it as-is for now.